### PR TITLE
Show the zero savings in object data reduction card

### PR DESCRIPTION
**Pull request by afreen23**
### Bug
mini2

### Fix
new2

Refactored the code a bit.
Also changed the units back to `withB` in order to be consistent with the console

### CODE REVIEW

**Comment by afreen23** [comment]
@cloudbehl please review

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 1]
```diff
@@ -59,28 +55,22 @@ export const SavingsItem: React.FC<SavingsItemProps> = React.memo(
...
-      </ItemBody>
-    );
+    return <ItemBody title="Savings" text={text} infoText={infoText} />;
   },
 );
 
 type ItemBodyProps = {
   title: string;
-  children: React.ReactNode;
+  text: React.ReactElement;
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]:
```suggestion
  text: string;
```

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 2]
```diff
@@ -59,28 +55,22 @@ export const SavingsItem: React.FC<SavingsItemProps> = React.memo(
     } else if (!_.isNil(savings)) {
       const savingsPercentage = logicalSize
         ? `(${humanizePercentage((100 * Number(savings)) / logicalSize).string})`
-        : null;
-      const savingsFormattted = humanizeBinaryBytesWithoutB(Number(savings));
+        : '';
+      const savingsFormattted = humanizeBinaryBytesWithoutB(savings).string;
       text = (
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]:
```suggestion
      text = `${savingsFormattted} ${savingsPercentage}`
```

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 3]
```diff
@@ -7,10 +7,13 @@ import {
   LoadingInline,
 } from '@console/internal/components/utils';
 
-const ItemBody: React.FC<ItemBodyProps> = React.memo(({ title, children }) => (
+const ItemBody: React.FC<ItemBodyProps> = React.memo(({ title, text, infoText }) => (
   <div className="co-inventory-card__item">
     <div className="nb-object-data-reduction-card__row-title">{title}</div>
-    {children}
+    <div className="nb-object-data-reduction-card__row-status-item">
+      {text}
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]:
```
<span className="nb-object-data-reduction-card__row-status-item-text">
          {text}
</span>
```

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 4]
```diff
@@ -59,28 +55,22 @@ export const SavingsItem: React.FC<SavingsItemProps> = React.memo(
     } else if (!_.isNil(savings)) {
       const savingsPercentage = logicalSize
         ? `(${humanizePercentage((100 * Number(savings)) / logicalSize).string})`
-        : null;
-      const savingsFormattted = humanizeBinaryBytesWithoutB(Number(savings));
+        : '';
+      const savingsFormattted = humanizeBinaryBytesWithoutB(savings).string;
```
Comment (gnehapk) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: Typo `savingsFormatted`

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx (hunk 3 above)
Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: `text` is `Unavailable` and  `<LoadingInline/>`  and they have custom css as well
I would rather generalize them all up!
Thanx!

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: same as above

**Comment by afreen23** [comment]
Generaized the `ItemBody` now , have a look @cloudbehl

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 5]
```diff
@@ -2,85 +2,60 @@ import * as React from 'react';
 import * as _ from 'lodash';
 import { DashboardCardHelp } from '@console/internal/components/dashboard/dashboard-card';
 import {
-  humanizeBinaryBytesWithoutB,
+  humanizeBinaryBytes,
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: Is it fine to change it here ?
#2296 is going in or not ?
@cloudbehl

Comment (cloudbehl) [reply frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: @afreen23 I don't think so.

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 6]
```diff
@@ -7,80 +7,55 @@ import {
...
 } from '@console/internal/components/utils';
 
-const ItemBody: React.FC<ItemBodyProps> = React.memo(({ title, children }) => (
-  <div className="co-inventory-card__item">
-    <div className="nb-object-data-reduction-card__row-title">{title}</div>
-    {children}
-  </div>
-));
+const ItemBody: React.FC<ItemBodyProps> = React.memo(({ title, stats, infoText, isLoading }) => {
+  let status = <span className="nb-object-data-reduction-card__row-status-item-text">{stats}</span>;
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: Type of status not defined.

Comment (cloudbehl) [reply frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: Better handle this in else condition.

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx [hunk 7]
```diff
@@ -7,80 +7,55 @@ import {
...
-        Unavailable
-      </span>
-    );
     const infoText =
       'Savings shows the uncompressed and non-deduped data that would have been stored without those techniques';
-
-    if (isLoading) {
-      text = <LoadingInline />;
-    } else if (!_.isNil(savings)) {
+    let stats = null;
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: Type not defined.

File: frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx (hunk 6 above)
Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: Why to handle in else condition ?

Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/object-data-reduction-card/object-data-reduction-card-item.tsx]: okay got it!

**Comment by cloudbehl** [comment]
@afreen23 please squash the commits.

**Pull request merged**